File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    var matrix = matrix
    var dimension = matrix.length;
    var row = [];
    var column = [];
    
    for(let i = 0; i < dimension; i++){
        for( let j = 0; j < matrix[0].length; j++){
            if(matrix[i][j] == 0){
                row.push(i);
                column.push(j);
            }
        }
    }
    
    for(let i = 0; i < row.length; i++){
        var changeRowLength = matrix[row[i]].length;
        var index = row[i]
        matrix[index] = []
        for(let j = 0; j < changeRowLength ; j++){
            matrix[index].push(0)
        }
    }
    
    for(let i = 0; i< dimension; i++){
        for(let j = 0; j < column.length; j++){
            var columnIndex = column[j]
            matrix[i][columnIndex] = 0;
        }
    }
    
    return (matrix);
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int col0 = 1;
        int row = matrix.length; 
        int cols = matrix[0].length;

        for(int i = 0; i < row; i++){
            if(matrix[i][0] == 0) col0 = 0;

            for(int j = 1; j < cols; j++)
            if(matrix[i][j] == 0)
            matrix[i][0] = matrix[0][j] = 0;
        }   

        for(int i = row - 1; i >= 0; i--){
            for(int j = cols - 1; j >= 1; j--)
            if(matrix[i][0] == 0 || matrix[0][j] == 0)
            matrix[i][j] = 0;

            if(col0 == 0) matrix[i][0] = 0;
        }
    }
}